Planit/Model/plan.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  plan.swift
+//  Planit
+//
+//  Created by Shivangi on 05/12/22.
+//
+
+import Foundation
+import SwiftUI
+import FirebaseFirestoreSwift
+
+struct plan: Identifiable, Decodable{
+    @DocumentID var id: String? = UUID().uuidString
+    let ownerUid: String
+    let title: String
+    let description: String
+    let planType: String
+    var completed: Bool
+    var documentID: String? = ""
+    
+    func getType() -> SelectedButton {
+        if planType == "itenary"{
+            return .Itenary
```

Comment:
A nice feature with Swift enums is that you can initialize them from their backing value, which is called `rawValue`. So you could use `SelectedButton(rawValue: planType)` to get the type dynamically, and then fallback on `.extra` if it's nil.